lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -180,7 +181,7 @@ impl<TMI: TagMetadataInterceptor> DogstatsdCodec<TMI> {
 
         events.push(Event::Metric(Metric::from_parts(context, values, metadata)));
 
-        Ok(1)
+        Ok(values_len)
```

Comment:
We actually want to emit the number of samples as the number of events because, ultimately, "metrics received" for the DogStatsD source should be able to speak to single _and_ multi-value payloads.